protocol/amqp/v2/protocol.go

```diff
@@ -92,69 +101,76 @@ func NewProtocol(server, queue string, connOption []amqp.ConnOption, sessionOpti
 }
 
 // NewSenderProtocolFromClient creates a new amqp sender transport.
-func NewSenderProtocolFromClient(client *amqp.Client, session *amqp.Session, address string, opts ...Option) (*Protocol, error) {
+func NewSenderProtocolFromClient(
+	ctx context.Context,
+	client *amqp.Conn,
+	session *amqp.Session,
+	address string,
+	opts ...Option,
+) (*Protocol, error) {
 	t := &Protocol{
-		Node:             address,
-		senderLinkOpts:   []amqp.LinkOption(nil),
-		receiverLinkOpts: []amqp.LinkOption(nil),
-		Client:           client,
-		Session:          session,
+		Node:    address,
+		Client:  client,
+		Session: session,
 	}
 	if err := t.applyOptions(opts...); err != nil {
 		return nil, err
 	}
-	t.senderLinkOpts = append(t.senderLinkOpts, amqp.LinkTargetAddress(address))
+
 	// Create a sender
-	amqpSender, err := session.NewSender(t.senderLinkOpts...)
+	amqpSender, err := session.NewSender(ctx, address, t.senderOpts)
 	if err != nil {
 		_ = client.Close()
 		_ = session.Close(context.Background())
 		return nil, err
 	}
-	t.Sender = NewSender(amqpSender).(*sender)
+	t.Sender = NewSender(amqpSender, t.sendOpts).(*sender)
 	t.SenderContextDecorators = []func(context.Context) context.Context{}
 
 	return t, nil
 }
 
 // NewReceiverProtocolFromClient creates a new receiver amqp transport.
-func NewReceiverProtocolFromClient(client *amqp.Client, session *amqp.Session, address string, opts ...Option) (*Protocol, error) {
+func NewReceiverProtocolFromClient(
+	ctx context.Context,
+	client *amqp.Conn,
+	session *amqp.Session,
+	address string,
+	opts ...Option,
+) (*Protocol, error) {
 	t := &Protocol{
-		Node:             address,
-		senderLinkOpts:   []amqp.LinkOption(nil),
-		receiverLinkOpts: []amqp.LinkOption(nil),
-		Client:           client,
-		Session:          session,
+		Node:    address,
+		Client:  client,
+		Session: session,
 	}
 	if err := t.applyOptions(opts...); err != nil {
 		return nil, err
 	}
 
 	t.Node = address
-	t.receiverLinkOpts = append(t.receiverLinkOpts, amqp.LinkSourceAddress(address))
-	amqpReceiver, err := t.Session.NewReceiver(t.receiverLinkOpts...)
+	amqpReceiver, err := t.Session.NewReceiver(ctx, address, t.receiverOpts)
 	if err != nil {
 		return nil, err
 	}
-	t.Receiver = NewReceiver(amqpReceiver).(*receiver)
+	t.Receiver = NewReceiver(amqpReceiver, t.receiveOpts).(*receiver)
 	return t, nil
 }
 
 // NewSenderProtocol creates a new sender amqp transport.
-func NewSenderProtocol(server, address string, connOption []amqp.ConnOption, sessionOption []amqp.SessionOption, opts ...Option) (*Protocol, error) {
-	client, err := amqp.Dial(server, connOption...)
+func NewSenderProtocol(ctx context.Context, server, address string, connOptions amqp.ConnOptions, sessionOptions amqp.SessionOptions, opts ...Option) (*Protocol, error) {
```

Comment:
`opts` also seems to be unused? (seems this was the case before?)

Reply:
those are used on line 173 for the internal protocol creation